src/backend/metal/src/factory.rs

```diff
@@ -150,6 +151,64 @@ impl Factory {
 
         self.device.new_depth_stencil_state(desc)
     }
+
+    pub fn create_library<P: AsRef<Path>>
+        (&mut self,
```

Comment:
Sorry, I was replicating what was done elsewhere in the file (e.g. Factory::create_buffer_immutable_raw on line 226) when there was a long return type that wouldn't fit if the usual line breaking (i.e. with &mut self on the same line) wouldn't work. How should I format it?